### ISSUE TITLE
Unit tests in contract.test.ts are failing 

### ISSUE THREAD

**Issue by ZeRego** [Loopring/loopring_sdk#10]
**Reproduction steps:**
`yarn test` 

**Expected:**
All tests pass

**Actual:**
Only the first test to run passes

**Investigation:** 
(Note that I'm a beginner is eth/loopring/crypto so I might be using the wrong terms.)

First problem:
We don't await for transactions to be part of a block so the _nonce_ value is the same for all the unit tests. 
This means only the first unit test to run passes and the rest throw an error. 

Possible solutions:
- We can await for the transaction to have a hash blockHash.
Example of util function:
```
async function waitForTransactionToBeAcknowledged(transactionHash: string) {
    const transaction = await web3.eth.getTransaction(transactionHash);
    if (transaction.blockHash) {
        return;
    } else {
        await waitForTransactionToBeAcknowledged(transactionHash);
    }
}
```

- We can get the nonce once and increment the value per each unit test.
```
    let nonce: number = 0;
    beforeAll(async() => {
        nonce = await contract.getNonce(web3, acc.address)
    })

    beforeEach(() => {
        nonce= nonce + 1;
    })
```

Second problem:

  contract test › deposit_LRC2 is throwing the error:
    _Cannot convert string to buffer. toBuffer only supports 0x-prefixed hex strings and this string was given: 0_

I haven't spent time debugging this issue but looks like a legit bug.

**Comment by brettwilcox** [comment]
We probably need to setup a .github folder with a github action to run the tests and report back.

[I propose we add Jest](https://jestjs.io/) for project unit testing.

**Comment by isaacaskew** [comment]
Ran this as well and can confirm that most tests do pass individually but not when all tests are run; seems to be just a test setup problem with the nonces that @ZeRego mentioned.

I got the same error with `Cannot convert string to buffer. toBuffer only supports 0x-prefixed hex strings and this string was given: 0`; I found a similar issue here: https://ethereum.stackexchange.com/questions/80205/cannot-convert-string-to-buffer-tobuffer-only-supports-0x-prefixed-hex-strings

Looks like we require the `contract.deposit` method to provide a value that is a `number`:

```
    value: number,
```

But `signEthereumTx`, called by `contract.deposit` -> `sendRawTx` -> `signEthereumTx`, expects `value` to be a hex string.

This makes me think that this line in `contract_api.ts` on line 198 would also fail, if there were a test for it:

```
    return await sendRawTx(web3, from, to, '0', data, chainId, nonce, gasPrice, gasLimit, sendByMetaMask)
```
